SimTKcommon/Simulation/include/SimTKcommon/internal/State.h

```diff
@@ -812,6 +813,11 @@ inline int getNEventTriggersByStage(Stage) const;
 /// first of the event triggers associated with a particular Stage are stored;
 /// the rest follow contiguously. Callable at Instance stage.
 inline SystemEventTriggerIndex getEventTriggerStartByStage(Stage) const;
+// Returns a pointer to the mutex stateLock that should be used
+// whenever multiple threads are asynchronously writing/updating a common state
+// cache. If multiple threads are simply reading from the cache, locking the state
+// may not be necessary
+inline std::mutex* getStateLock() const;
```

Comment:
- This is part of the State API so it needs doxygen comments (triple slash)
- Why is this returning a pointer to a mutex rather than a reference? Pointers should only be used when `nullptr` is a possible value, which I don't think is your intention here.